etcdserver/v3_server.go

```diff
@@ -533,6 +562,11 @@ func (s *EtcdServer) raftRequestOnce(ctx context.Context, r pb.InternalRaftReque
 	if result.err != nil {
 		return nil, result.err
 	}
+	if startTime, ok := ctx.Value("time").(time.Time); ok && result.trace != nil {
+		applyStart := result.trace.ResetStartTime(startTime)
```

Comment:
Can we add comments to explain why resetting the start time? I think I understand why: trace is created in apply, but we actually want to trace the steps before apply w/o piping trace object through the entire raft package. But let's document it so it is more clear.